src/DataProtection/DataProtection/test/Microsoft.AspNetCore.DataProtection.Tests/KeyManagement/KeyRingProviderTests.cs

```diff
@@ -718,46 +718,6 @@ public void GetCurrentKeyRing_NoExistingKeyRing_HoldsAllThreadsUntilKeyRingCreat
         mockCacheableKeyRingProvider.Verify(o => o.GetCacheableKeyRing(It.IsAny<DateTimeOffset>()), Times.Once);
     }
 
-    [Fact]
-    public void GetCurrentKeyRing_WithExpiredExistingKeyRing_AllowsOneThreadToUpdate_ReturnsExistingKeyRingToOtherCallersWithoutBlocking()
-    {
-        // Arrange
-        var originalKeyRing = new Mock<IKeyRing>().Object;
-        var originalKeyRingTime = StringToDateTime("2015-03-01 00:00:00Z");
-        var updatedKeyRing = new Mock<IKeyRing>().Object;
-        var updatedKeyRingTime = StringToDateTime("2015-03-02 00:00:00Z");
-        var mockCacheableKeyRingProvider = new Mock<ICacheableKeyRingProvider>();
-        var keyRingProvider = CreateKeyRingProvider(mockCacheableKeyRingProvider.Object);
-
-        // In this test, the foreground thread acquires the critial section in GetCurrentKeyRing,
-        // and the background thread returns the original key ring rather than blocking while
-        // waiting for the foreground thread to update the key ring.
-
-        TimeSpan testTimeout = TimeSpan.FromSeconds(10);
-        IKeyRing keyRingReturnedToBackgroundThread = null;
-
-        mockCacheableKeyRingProvider.Setup(o => o.GetCacheableKeyRing(originalKeyRingTime))
-            .Returns(new CacheableKeyRing(CancellationToken.None, StringToDateTime("2015-03-02 00:00:00Z"), originalKeyRing));
-        mockCacheableKeyRingProvider.Setup(o => o.GetCacheableKeyRing(updatedKeyRingTime))
-            .Returns<DateTimeOffset>(dto =>
-            {
-                // at this point we're inside the critical section - spawn the background thread now
```

Comment:
This test was unsalvageable, because it depended upon the lambda being evaluated under a particular lock.  `MultipleThreadsSeeExpiredCachedValue` attempts to provide analogous coverage.